server/src/main/java/org/elasticsearch/threadpool/ThreadPool.java

```diff
@@ -321,6 +330,60 @@ public ThreadPool(final Settings settings, final ExecutorBuilder<?>... customBui
         this.cachedTimeThread.start();
     }
 
+    private static ArrayList<Instrument> setupMetrics(MeterRegistry meterRegistry, String name, ExecutorHolder holder) {
+        Map<String, Object> at = Map.of();
+        ArrayList<Instrument> instruments = new ArrayList<>();
+        if (holder.executor() instanceof ThreadPoolExecutor threadPoolExecutor) {
+            String prefix = "es.thread_pool." + name + ".threads.";
+            logger.info("Starting tp metrics {}", name);
+            instruments.add(
+                meterRegistry.registerLongGauge(
+                    prefix + "count",
```

Comment:
can we use a `current` suffix here?

Reply:
Addressed in 1a0b1defba83ade4b673cf9af21fa09161397f50